### PR TITLE
runAsNonRoot true in readme

### PR DESCRIPTION
**Pull request by cicoyle**
runAsNonRoot true, fix table format in README, make modtidy-all

## Issue reference

Please reference the issue this PR will close: https://github.com/dapr/dapr/issues/7863

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: charts/dapr/charts/dapr_scheduler/templates/dapr_scheduler_statefulset.yaml [hunk 1]
```diff
@@ -164,7 +164,7 @@ spec:
         - "--mode=kubernetes"
 {{- if eq .Values.global.daprControlPlaneOs "linux" }}
         securityContext:
-          runAsUser: 0
+          runAsUser: {{ .Values.runAsNonRoot }}
```
Comment (joebowbeer) [inline charts/dapr/charts/dapr_scheduler/templates/dapr_scheduler_statefulset.yaml:167]:
```suggestion
          runAsNonRoot: {{ .Values.runAsNonRoot }}
```

**Review by joebowbeer — CHANGES_REQUESTED** [review]
Looks like a typo

runAsNotRoot and runAsUser are two different but related settings.

**Comment by cicoyle** [comment]
> Looks like a typo
> 
> runAsNotRoot and runAsUser are two different but related settings.

Good catch. Thank you!

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]
Changing `runAsNonRoot` to `false` is a really major change that comes with security implications. A lot of K8s users run scanners that will flag Dapr containers if they are running as root.

What's the reason behind the change? #7863 mentions a discrepancy between docs and YAML files. In this case, shouldn't the docs be updated first?

File: charts/dapr/charts/dapr_operator/values.yaml [hunk 2]
```diff
@@ -23,7 +23,7 @@ webhookService:
   type: ClusterIP
   annotations: {}
 
-runAsNonRoot: true
+runAsNonRoot: false
```
Comment (joebowbeer) [inline charts/dapr/charts/dapr_operator/values.yaml:26]:
```suggestion
runAsNonRoot: true
```

Suggest retaining true as it is more secure.

File: charts/dapr/charts/dapr_placement/values.yaml [hunk 3]
```diff
@@ -53,7 +53,7 @@ debug:
   port: 40000
   initialDelaySeconds: 30000
 
-runAsNonRoot: true
+runAsNonRoot: false
```
Comment (joebowbeer) [inline charts/dapr/charts/dapr_placement/values.yaml:56]:
```suggestion
runAsNonRoot: true
```

File: charts/dapr/charts/dapr_scheduler/values.yaml [hunk 4]
```diff
@@ -52,7 +52,7 @@ debug:
   port: 40000
   initialDelaySeconds: 30000
 
-runAsNonRoot: true
+runAsNonRoot: false
```
Comment (joebowbeer) [inline charts/dapr/charts/dapr_scheduler/values.yaml:55]:
```suggestion
runAsNonRoot: true
```

File: charts/dapr/charts/dapr_sentry/values.yaml [hunk 5]
```diff
@@ -44,7 +44,7 @@ debug:
   port: 40000
   initialDelaySeconds: 30000
 
-runAsNonRoot: true
+runAsNonRoot: false
```
Comment (joebowbeer) [inline charts/dapr/charts/dapr_sentry/values.yaml:47]:
```suggestion
runAsNonRoot: true
```

File: charts/dapr/charts/dapr_sidecar_injector/values.yaml [hunk 6]
```diff
@@ -25,7 +25,7 @@ nameOverride: ""
 fullnameOverride: ""
 webhookFailurePolicy: Ignore
 sidecarImagePullPolicy: IfNotPresent
-runAsNonRoot: true
+runAsNonRoot: false
```
Comment (joebowbeer) [inline charts/dapr/charts/dapr_sidecar_injector/values.yaml:28]:
```suggestion
runAsNonRoot: true
```

**Comment by cicoyle** [comment]
> comes with security implication

Yes, there is a discrepancy btw the docs and implementation details. I assumed to be more secure, and set `runAsRoot` to `false`.

**Review by joebowbeer — CHANGES_REQUESTED** [review]
Preferably `runAsNonRoot` should continue to default to true whenever possible as it is more secure.

I think it is the default values in the README that should be updated to match.

**Comment by ItalyPaleAle** [comment]
> > comes with security implication
> 
> Yes, there is a discrepancy btw the docs and implementation details. I assumed to be more secure, and set `runAsRoot` to `false`.

`runAsNonRoot` should be `true`. **That** is the most secure option. Some reading on K8s security [here](https://dev.to/aws-builders/best-practices-for-securing-kubernetes-deployments-1jg6)

**Comment by cicoyle** [comment]
> > > comes with security implication
> > 
> > 
> > Yes, there is a discrepancy btw the docs and implementation details. I assumed to be more secure, and set `runAsRoot` to `false`.
> 
> `runAsNonRoot` should be `true`. **That** is the most secure option. Some reading on K8s security [here](https://dev.to/aws-builders/best-practices-for-securing-kubernetes-deployments-1jg6)

My mind stopped reading the word `runAs...` and I was thinking I was setting `runAsRoot` false, but its `runAsNonRoot`, which should be true. Yes, you are correct :)

**Review by joebowbeer — COMMENTED** [review]
LGTM!

File: tests/apps/resiliencyapp/go.mod [hunk 7]
```diff
@@ -12,7 +12,7 @@ require (
 
 require (
 	github.com/google/uuid v1.6.0 // indirect
-	go.opentelemetry.io/otel v1.26.0 // indirect
+	go.opentelemetry.io/otel v1.27.0 // indirect
```
Comment (ItalyPaleAle) [inline tests/apps/resiliencyapp/go.mod:15]: Changes to go.mod files seem irrelevant to this PR

File: charts/dapr/README.md [hunk 8]
```diff
@@ -172,67 +172,67 @@ The Helm chart has the follow configuration options that can be supplied:
...
+| Parameter                                                 | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                            | Default                                                                                                                                                                                                                   |
```
Comment (ItalyPaleAle) [inline charts/dapr/README.md:210]: Please don't change things that are unrelated in this README. Adding all this whitespace actually makes reading the README _harder_ when not using the rendered markdown...

File: tests/apps/resiliencyapp/go.mod (hunk 7 above)
Comment (cicoyle) [reply tests/apps/resiliencyapp/go.mod:15]: if you look in the go.mod in dapr it has that version for otel, but it must have been missed for the ones in this PR. I ran `make modtidy-all`

File: charts/dapr/README.md (hunk 8 above)
Comment (cicoyle) [reply charts/dapr/README.md:210]: This diff makes the tables properly formatted and consistent. Some of the tables were properly aligned and the ones updated in this PR were not properly aligned, hence the diff. My IDE was complaining about the format and auto fixed it. If you have it locally it reads fine and the text is aligned well. I would think having everything properly aligned would lean towards being easier to digest.
